File: median-of-two-sorted-arrays/median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        n1 = len(nums1)
        n2 = len(nums2)
        medianIdx = (n1 + n2) // 2
        i = 0
        ptr1 = 0
        ptr2 = 0
        prevNum = 0
        num = 0
        while i <= medianIdx:
            prevNum = num
            if ptr2 >= n2:
                print("get num1")
                num = nums1[ptr1]
                ptr1 += 1
            elif ptr1 >= n1:
                print("get num2")
                num = nums2[ptr2]
                ptr2 += 1
            else:
                if nums1[ptr1] <= nums2[ptr2]:
                    num = nums1[ptr1]
                    ptr1 += 1
                else:
                    num = nums2[ptr2]
                    ptr2 += 1
            print("i = {}, num = {}".format(i, num))
            i += 1
                    
        # if odd
        if (n1 + n2) % 2 != 0:
            print("num = {}".format(prevNum, num))
            return num
        # if even
        print("prev = {}, num = {}".format(prevNum, num))

        return (num + prevNum) / 2
                
            

File: median-of-two-sorted-arrays/median-of-two-sorted-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    // find index of the median + 1 index
    const n = nums1.length,
          m = nums2.length;
    const endIndex = Math.min(((n + m - 1) >> 1) + 1, n+m-1);
    console.log(endIndex);
    let i = 0, j = 0;
    const merged = [];
    for (let idx = 0; idx <= endIndex; idx++){
        let valueA = Number.POSITIVE_INFINITY,
            valueB = Number.POSITIVE_INFINITY;
        if (i < n){
            valueA = nums1[i];
        }
        if (j < m){
            valueB = nums2[j];
        }
        // compare value
        // console.log([valueA, valueB])
        if (valueA <= valueB){
            merged.push(valueA);
            i++;
        }else{
            merged.push(valueB);
            j++;
        }
        if (merged.length > 2){
            merged.shift(); // pop first
        }
    }
    // console.log(merged)
    // test if odd or even
    if ((n + m) & 1 == 1){
        // odd
        return merged[0];
    }else{
        return (merged[0] + merged[1]) / 2;
    }
    
};